README.md

```diff
@@ -1,2 +1,288 @@
+
+> :memo: **Note:** [PID Notes](https://github.com/KudoC0nan/Balance_On_Wheels/blob/dev_arsh/Notes/PID%20Explanation)
+
+## <ins> Code Flow
+
+Here, we First wrote the algorithm for Self-Balancing the bot.
+
+
+For self-balancing the bot, we need to keep the bot a bit above our desired angle (Setpoint), since the bot is very fast and will fall down once it's past the desired angle.
+
+The pitch angle is required for Self-Balancing which is obtained from the Inertial Measurement Unit(IMU sensor) plug-in.
+
+> :memo: **Note:**[Link to Explanation of Self Balancing Code](https://github.com/KudoC0nan/Balance_On_Wheels/blob/dev_arsh/Notes/Self%20Balancing%20Code.md)
+
+
+And after Successful Run of Self Balancing Code, we than
+wrote the code for Servo moving from Maximum to Minimum Length (Variable Height).
+
+[Link to Code]()
+
+And after that we wrote the code of integrating Self Balancing Code with servos locking up at minimum height for Self Balancing of Bot at Minimum Height.
+
+[Link to Code](https://github.com/KudoC0nan/Balance_On_Wheels/tree/dev_arsh/Self_Balancing_Servo_set_zero)
+
+**Flow Chart of Code**
+![](https://i.imgur.com/6f7omtf.png)
+
+After Which we wrote the Jumping Code for Bot to Jump from Minimum Position.
+
+[Link to Code]()
+
+After Which we wrote the code for integration of Jumping Code with Self Balancing Code.
+
+[Link to Code]()
+
+---
+## <ins>Results and Demo
+
+--- 
+## <ins>Future Work
+1) Segway Bot , in which we can add camera ,speech recognition and self driving facility in a confined space. We can also add arms to it . So this Bot can interact with people (by recognizing their face and voice) ,can change height and  with arms can lift up objects.(Assistant Type)   
+We can also use DFS ans BFS algorithm.
+
+    https://techcrunch.com/2016/01/07/segway-has-created-a-robot-that-connects-to-your-two-wheeled-scooter/
+
+2) For obstacle avoiding , ultrasonic sensors (hc sr04) or  Lidar sensors or IR sensors along with our present bot can be used
+Lidar sensor has ability to measure 3d structures accurately .It uses laser beam.
+Cost is high 
+Ultrasonic sensors uses sound waves for detection. It's cost effective. Less sensor accuracy and limited detection range 
+
+    Sensor fusion will enables us to improve obstacle detection and localization of objects in dark and foggy areas 
+    Also more accurate estimation of environment (wind speed etc) 
+
+    Also for obstacle detection we can use computer vision also.
+    https://www.science.gov/topicpages/o/obstacle+detection+algorithm
+(2nd and 3rd paper algorithm is useful for us) 
+
+3) Multi Terrain RoBot which can change the length of individual legs based upon terrain and balance itself accordingly
+
+---
+
+## <ins>Troubleshooting
+
+1. First design had Curves which reduced the integrity of the structure which were then removed in the 2nd Iteration.
```

Comment:
First Point was not an issue for you. You guys had directly started with links without curves

You didnt explain mounting of motors at wheel joint, please think and add issue which you had faced.

Reply:
made changes and added more issues!